stats/tally.go

```diff
@@ -0,0 +1,188 @@
+package stats
```

Comment:
Might be worth stuttering this package name, e.g., `tchanneltally`.

Reply:
This is adding to the existing tchannel stats package:
https://godoc.org/github.com/uber/tchannel-go/stats